test/support/integration_test.rb

```diff
@@ -16,6 +16,10 @@ def initialize(args)
   end
 
   def setup
+    # search_config is a global object that has state, so make sure it's reset
```

Comment:
It might be nice to split `SearchConfig` up at some point: it's a bit surprising (to me at least 🙂) that it holds state like the registries which contain data from Elasticsearch rather than static config.